### ISSUE TITLE
Resources not closed in reversed order

### ISSUE THREAD

**Issue by Torehs** [kotest/kotest#1382]
We recently upgraded kotest to version 4.0.2 (from 3.3.1 previously). According to the [documentation](https://github.com/kotest/kotest/blob/master/doc/reference.md#closing-resource-automatically) closing is performed in reversed order of declaration after the return of the last spec interceptor when using autoClose. This was the behavior we had before the upgrade. Now it looks like resources are closed in order of declaration and not the reversed order. Is that a bug or a change in behavior?

I used this code for my testing:
```
class Object(val version: Int) : Closeable {
    init {
        println("Object $version created")
    }
    override fun close() {
        println("Object $version deleted")
    }
}

class Test : StringSpec({
    "Test" {
        autoClose(Object(1))
        autoClose(Object(2))
    }
})
```
Expected output:
```
Object 1 created
Object 2 created
Object 2 deleted
Object 1 deleted
```

Actual output:
```
Object 1 created
Object 2 created
Object 1 deleted
Object 2 deleted
```
Let me know if you need more information from me.

**Comment by sksamuel** [comment]
Will be included in 4.0.3

**Comment by Torehs** [comment]
Thanks a lot!
